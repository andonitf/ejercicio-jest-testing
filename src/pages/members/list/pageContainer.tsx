import * as React from 'react';
import { MemberListPage } from './page';
import { Member } from './viewModel';
import { fetchMembers } from '../../../rest-api/api/member';
import { State } from '../../reducers';
import { getMembersVM } from './selectors';
import { connect } from 'react-redux';

const mapStateToProps = (state: State) => ({
  members: getMembersVM(state),
});
const mapDispatchToProps = (dispatch) => ({
  fetchMembers: () => dispatch(fetchMembers()),
});

interface Props {
  members: Member[];
  fetchMembers: () => void;
}
class PageContainer extends React.PureComponent<Props, {}> {
  componentDidMount() {
    this.props.fetchMembers();
  }

  render() {
    return (
      <MemberListPage
        members={this.props.members}
      />
    );
  }
}
export const MemberListPageContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(PageContainer);
