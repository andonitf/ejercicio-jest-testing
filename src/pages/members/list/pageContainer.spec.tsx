import * as React from 'react';
import { shallow } from 'enzyme';
import { MembersState } from './reducers';
import { MemberListPageContainer } from './pageContainer';
import configureStore from 'redux-mock-store';

const getMockStore = configureStore();

describe('pages/members/list/pageContainer specs', () => {
  it('should render as expect when passing state', () => {
    // Arrange
    const state: MembersState = [
      { id: 1, login: 'Login 1', avatar_url: 'http://fake-test-url/member/1' },
      { id: 2, login: 'Login 2', avatar_url: 'http://fake-test-url/member/2' },
      { id: 3, login: 'Login 3', avatar_url: 'http://fake-test-url/member/3' },
      { id: 4, login: 'Login 4', avatar_url: 'http://fake-test-url/member/4' },
      { id: 4, login: 'Login 5', avatar_url: 'http://fake-test-url/member/5' },
  ];

    const store = getMockStore(state);

    // Act
    const component = shallow(<MemberListPageContainer />, {
      context: { store },
    });

    // Assert
    expect(component).toMatchSnapshot();
  });
});
