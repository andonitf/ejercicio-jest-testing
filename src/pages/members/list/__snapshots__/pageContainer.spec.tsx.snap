// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/pageContainer specs should render as expect when passing state 1`] = `
<PageContainer
  fetchMembers={[Function]}
  members={Array []}
/>
`;
